src/api_v2/models/mod.rs

```diff
@@ -0,0 +1,35 @@
+mod user;
+mod beatmap;
+mod beatmapset;
+mod covers;
+
+use chrono::DateTime;
+use serde::{Deserialize, Deserializer};
+
+
+pub use user::{User, UserSave};
+pub use beatmap::BeatMap;
+pub use beatmap::Failtimes;
+pub use beatmapset::BeatMapSet;
+pub use covers::Covers;
+
+
+pub fn s_to_datetime<'de, D>(deserializer: D) -> Result<Option<i64>, D::Error>
+    where
+        D: Deserializer<'de>,
+{
+    let p = String::deserialize(deserializer);
+    if p.is_err() {
```

Comment:
This can be replaced by `.ok()` method: https://doc.rust-lang.org/std/result/enum.Result.html#method.ok

Reply:
like this?

```
let p = p.ok();
    if p.is_none() {
        return Ok(None);
    }
    let p = p.unwrap();
```

Reply:
Umm it seems like I misunderstand your proposal. The code is fine.